ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/services/UserManagementServiceAuthorization.scala

```diff
@@ -28,60 +27,66 @@ private[daml] final class UserManagementServiceAuthorization(
+      case Right(None) => authorizer.requireAdminClaims(service.getUser)(request)
     }
 
-  override def deleteUser(request: DeleteUserRequest): Future[DeleteUserResponse] =
+  override def deleteUser(request: proto.DeleteUserRequest): Future[proto.DeleteUserResponse] =
     authorizer.requireAdminClaims(service.deleteUser)(request)
 
-  override def listUsers(request: ListUsersRequest): Future[ListUsersResponse] =
+  override def listUsers(request: proto.ListUsersRequest): Future[proto.ListUsersResponse] =
     authorizer.requireAdminClaims(service.listUsers)(request)
 
-  override def grantUserRights(request: GrantUserRightsRequest): Future[GrantUserRightsResponse] =
+  override def grantUserRights(
+      request: proto.GrantUserRightsRequest
+  ): Future[proto.GrantUserRightsResponse] =
     authorizer.requireAdminClaims(service.grantUserRights)(request)
 
   override def revokeUserRights(
-      request: RevokeUserRightsRequest
-  ): Future[RevokeUserRightsResponse] =
+      request: proto.RevokeUserRightsRequest
+  ): Future[proto.RevokeUserRightsResponse] =
     authorizer.requireAdminClaims(service.revokeUserRights)(request)
 
-  override def listUserRights(request: ListUserRightsRequest): Future[ListUserRightsResponse] =
+  override def listUserRights(
+      request: proto.ListUserRightsRequest
+  ): Future[proto.ListUserRightsResponse] =
     defaultToAuthenticatedUser(request.userId) match {
-      case Failure(ex) => Future.failed(ex)
-      case Success(Some(userId)) => service.listUserRights(request.copy(userId = userId))
-      case Success(None) => authorizer.requireAdminClaims(service.listUserRights)(request)
+      case Left(e: StatusRuntimeException) => Future.failed(e)
+      case Right(Some(userId)) => service.listUserRights(request.copy(userId = userId))
+      case Right(None) => authorizer.requireAdminClaims(service.listUserRights)(request)
     }
 
-  override def updateUser(request: UpdateUserRequest): Future[UpdateUserResponse] = {
+  override def updateUser(request: proto.UpdateUserRequest): Future[proto.UpdateUserResponse] = {
     authorizer.requireAdminClaims(service.updateUser)(request)
   }
 
   override def bindService(): ServerServiceDefinition =
-    UserManagementServiceGrpc.bindService(this, executionContext)
+    proto.UserManagementServiceGrpc.bindService(this, executionContext)
 
   override def close(): Unit = service.close()
 
-  private def defaultToAuthenticatedUser(userId: String): Try[Option[String]] =
+  private def defaultToAuthenticatedUser(
+      userId: String
+  ): Either[StatusRuntimeException, Option[String]] =
     authorizer.authenticatedUserId().flatMap {
       case Some(authUserId) if userId.isEmpty || userId == authUserId =>
         // We include the case where the request userId is equal to the authenticated userId in the defaulting.
-        Success(Some(authUserId))
+        Right(Some(authUserId))
```

Comment:
Using `proto.` imports above because otherwise `proto.Right` what shadow the scala's `Right`